### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by mnajdova**
Addressing comment https://github.com/mui-org/material-ui/pull/23370#pullrequestreview-524156204

### CODE REVIEW

File: docs/scripts/buildApi.ts [hunk 1]
```diff
@@ -192,78 +192,31 @@ async function annotateComponentDefinition(context: {
...
-          });
-
-          api.styles.classes = classes;
-        }
-      },
-    });
+  const componentName = path.basename(component.filename).replace(/\.js$/, '');
+  const cssFilename = `../data/css/${camelCaseToKebabCase(componentName)}-css.js`;
+  try {
+    const cssModule = require(cssFilename);
```
Comment (eps1lon) [inline docs/scripts/buildApi.ts:206]: I'd just read the json with `fs-extra`. `require` is slowly getting replace and with require we have to rely on webpack.

**Review by eps1lon — CHANGES_REQUESTED** [review]
The section would still require an overhaul since it contains outdated/misleading information from the `classes` API.

**Comment by mnajdova** [comment]
> The section would still require an overhaul since it contains outdated/misleading information from the classes API.

These classes are still used in the theme overrides as keys, it is not misleading information. We just don't support the `classes` prop

File: docs/scripts/buildApi.ts (hunk 1 above)
Comment (mnajdova) [reply docs/scripts/buildApi.ts]: Converted the file to `.json` and used `readFileSync`. Should we merge this on the main branhc?

**Comment by eps1lon** [comment]
> These classes are still used in the theme overrides as keys, it is not misleading information. We just don't support the classes prop

How is it not misleading to mention the `classes` prop if it's not implemented?

**Comment by mnajdova** [comment]
> How is it not misleading to mention the classes prop if it's not implemented?

The classes keys are still used as keys in the styleOverrides inside the theme. Where am I mentioning the classes prop?

**Pull request merged**